src/UI/UIView.cpp

```diff
@@ -45,596 +54,905 @@ void UIView::draw_ui_callback()
-          tfn_widget_multi[m_table_selection].draw_ui();
+          else
+          {
+            m_simulation_state_selection--;
+            SimulationState next_sim_state = m_controller_app.get_simulation().get_simulation_state_at(m_simulation_state_selection);
+            m_controller_app.get_trackball_camera().set_current_poi(next_sim_state.poi);
+          }
+
+
         }
-        else
+
+      }
+      ImGui::SameLine();
+      if (ImGui::Button("Reset Camera")) {
+        m_controller_app.get_trackball_camera().reset_camera();
+      }
+      ImGui::BeginTable("##Camera Point of interest (POI) Editor", 1, ImGuiTableFlags_Borders);
+      ImGui::TableSetupColumn("Name", ImGuiTableColumnFlags_WidthFixed, 120.0f);
+
+
+      for (int row = 0; row < m_controller_app.get_simulation().get_simulation_states().size(); ++row)
+      {
+        ImGui::TableNextRow();
+        ImGui::TableSetColumnIndex(0);
+        bool item_is_selected = (row == m_simulation_state_selection) ? true : false;
+        SimulationState sim_state = m_controller_app.get_simulation().get_simulation_state_at(row);
+        if (ImGui::Selectable(sim_state.poi.label.c_str(), item_is_selected, ImGuiSelectableFlags_AllowDoubleClick))
         {
-          if (m_animated)
+          m_simulation_state_selection = row;
+          m_controller_app.get_trackball_camera().set_current_poi(sim_state.poi);
+          m_clip_min = sim_state.min_clip;
+          m_clip_max = sim_state.max_clip;
+          if (ImGui::IsMouseDoubleClicked(0))
           {
-            /*
-             frame by frame animation
+            m_copy_camera_name = sim_state.poi.label;
+            m_camera_name_window_open = true;
+            m_camera_button_action = BUTTON_ACTION::EDIT;
+          }
+        }
+      }
+      ImGui::EndTable();
 
-            unsigned int active_volume = floor(m_frame);
-              unsigned int active_volume2 = ceil(m_frame);
-              double alpha = m_frame - active_volume;
-              tfn_widget[m_selectedVolume].setMinMax(m_volumes[m_selectedVolume][active_volume]->getMin() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMin() * (1.0 - alpha),
-                m_volumes[m_selectedVolume][active_volume]->getMax() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMax() * (1.0 - alpha));
-              if (active_volume < m_volumes[m_selectedVolume].size() && active_volume2 < m_volumes[m_selectedVolume].size())
-                tfn_widget[m_selectedVolume].setBlendedHistogram(m_volumes[m_selectedVolume][active_volume]->getTransferfunction(0), m_volumes[m_selectedVolume][active_volume2]->getTransferfunction(0), alpha);*/
+      if (m_controller_app.get_simulation().get_simulation_states().size() > 0)
+      {
+        SimulationState& sim_state = m_controller_app.get_simulation().get_simulation_state_at(m_simulation_state_selection);
+        std::string edit_text = "Edit Time Frame " + sim_state.poi.label;
```

Comment:
what does time frame mean in this context?

Reply:
Time frame maps to a state of a simulation. A frame where the simulation has a specific camera pos and slice values.